### ISSUE TITLE
Favorites are shown in search even when they (no longer) match search text

### ISSUE THREAD

**Issue by mnalis** [osmandapp/OsmAnd#12111]
### Description

When searching by using name,  `Favorites` are displayed even if their name no longer matches name being typed (but did when we started typing).

### How to reproduce?
- click `Search` (looking glass icon)
- type `trg` and it matches (among other things) 3 Favorites (yellow star) which is OK (`Mažuranac - trg Mažuranića`, `Osijek - Tveđa, Trg.sv.trojstva` and `trg bana jelačića`)
- continue typing `trg stjepana`
  ![s_Screenshot_20210704_134953_net osmand plus](https://user-images.githubusercontent.com/156656/125684093-2ba519a1-c54b-481a-bf3d-63543b328c0b.jpg)
  and it continues showing those 3 Favorites, although none of them matches new search text now.
  I would expect that Favorites are **removed from search results when they no longer match search text** (as other matches which are note Favorites are removed correctly).

### Your Environment
OsmAnd Version: 3.9.10 (f-droid)
Android/iOS version: Android 10 (EMUI 11.0.0)
Device model: Huawei P30 Pro

**Maps used (online or offline):**   
Standard maps (vector): Croatia (Jul 1, 2021)

**Comment by Michael21011** [comment]
@mnalis if you will continue typing and enter trj stjepana severa I suppose your favorites will gone from search result. It is because until full equal of search string will not found then osmand will show partial result for every word in your search sentence. It works for all found record but  favorites have priority in sorting order, that is why you see it on the top.

**Comment by mnalis** [comment]
Thanks for your comment @Michael21011
> if you will continue typing and enter trj stjepana severa I suppose your favorites will gone from search result

Nope, even when I type full `trg stjepana severa` it still shows those 3 Favorites which do not match:
![s_Screenshot_20210716_055002_net osmand plus](https://user-images.githubusercontent.com/156656/125889150-1df333f7-ccb6-444d-ad2c-f083a917340a.jpg)

I've tried to verify your theory (that OsmAnd will try using ANY word instead of ALL words, if it cannot find match for ALL words), but it does not seem to be how it works in reality.

- for example when I try adding nonexistent third word after `trg stjepana`, it still finds `trg stjepana severa`, but only if that word is shorter or equal to 8 characters (1st picture below). 
- If I add just one letter to that nonexistent 3rd word, it suddenly stops finding results in POIs, but still "finds" them in Favorites (2nd pic below)
- In 3rd pic I've tried entering two words `tveđa` (present in one Favorite) and `mažuranac` (present in other Favorite). If your guess that it would search in Favorites for either `tveđa` **or** `mažuranac` was correct, then it should have found both Favorites, right? However, it finds only one Favorite (and one POI).
 
It all looks very confusing and nondeterministic, so I do not think that is intended behaviour.

 ![s_Screenshot_20210716_060056_net osmand plus](https://user-images.githubusercontent.com/156656/125889839-ba2b22d9-5a60-4562-829b-0158af257a37.jpg) ![s_Screenshot_20210716_060124_net osmand plus](https://user-images.githubusercontent.com/156656/125889840-7f88bcd5-0b98-48c8-a1cf-5d4fb1273d93.jpg) ![s_Screenshot_20210716_060459_net osmand plus](https://user-images.githubusercontent.com/156656/125890069-c2e6e990-eb0c-4180-9be6-9ffda2e3aa20.jpg)
